### PR TITLE
CLI: better existence check

### PR DESCRIPTION
**Pull request by chabou**
NPM registry is broken for some (unpublished?) package:
https://registry.npmjs.org/hyper-cwd
https://api.npms.io/v2/package/hyper-cwd (error 500)

Checking headers on npm is not enough.

Now, `versions` key existence is checked in content

### CODE REVIEW

File: cli/api.js [hunk 1]
```diff
@@ -62,14 +63,19 @@ function save() {
...
 function existsOnNpm(plugin) {
-  plugin = plugin.split('#')[0].split('@')[0];
-  return npmName(plugin).then(unavailable => {
-    if (unavailable) {
+  const name = plugin.split('#')[0].split('@')[0];
+  return got
+    .get(registryUrl + name.toLowerCase(), {timeout: 10000, json: true})
+    .then(res => {
+      if (!res.body.versions) {
+        throw new Error();
```
Comment (ppot) [inline cli/api.js]: Error should be defined with a message.

Comment (brandon93s) [reply cli/api.js]: Meh, just used to get into the `.catch` block here. Could alternatively `return Promise.reject();`.

Comment (ppot) [reply cli/api.js]: Your ready to go. 🚀

**Comment by chabou** [comment]
@brandon93s @ppot Thank you for your reviews. You were both right. I refactored it and looks a way better now.
Bonus: There is a convenient message for errors.

**Pull request merged**